lib/retriable.rb

```diff
@@ -6,14 +6,22 @@
 module Retriable
   module_function
 
-  def self.configure
+  def configure
     yield(config)
   end
 
   def config
     @config ||= Config.new
   end
 
+  def retriable_with_context(context_key, options = {}, &block)
```

Comment:
If I had to write it again I might leave out the kernel extension, but I don't want to break it for people upgrading. I changed it to name it `with_context` until you use the kernel extension, then it's `retriable_with_context`.